### PR TITLE
Web Client dialog tutorials

### LINKED ISSUE #92: Feature Request (missing/incomplete docs): Showing a dialog with telestion-client-common
### Type(/-s) of missing documentation

- [ ] Explanation of a concept
- [X] Tutorial for achieving a task with the Telestion ecosystem
- [ ] Guide that explains how to use a specific tool in more detail
- [ ] Reference to quickly look something up

### Description of missing documentation

As a Telestion Project developer, I want to show dialogs for specific UI cases.

It's great that `telestion-client` provides abstraction layers for achieving this, but the surrounding structure is pretty big (offering a lot of options, but also a lot of exported interfaces, etc.)

I would therefore like to see an easy step by step tutorial that shows

1. How to show a simple dialog
2. How to add form controls/state/return values to that dialog

This can either be
1. one single tutorial ("Showing dialogs") that "expands" the dialog created in the first steps with form controls in later steps of the tutorial
2. _or, better:_ Two tutorials with "Showing a dialog" linking to "Showing a dialog with a form" linked to it as "Next steps"

### Additional comments

Relates to https://github.com/wuespace/telestion-client/pull/855 as a prerequisite of a library release

### PR DESCRIPTION
**Pull request by fussel178**
### Summary 

Adds tutorials how to add static and stateful dialogs to your web client.

### Details 

n/a

### Additional information 

n/a

### Related links 

- Fixes #92 

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### CODE REVIEW

**Review by pklaschka — CHANGES_REQUESTED** [review]
Overall, great tutorials. I've left a few comments here and there, but apart from these minor change requests, it looks awesome!

Added via Giphy

File: docs/client/tutorials/301-writing-a-stateful-dialog.mdx [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+
+To complete this tutorial, you should be familiar with TypeScript language
+and the React framework. This tutorials builds upon the code from the simple dialog tutorial.
+
+:::
+
+## What we'll build
+
+In this tutorial, we'll extend the functionality from the simple dialog of the last tutorial
+and add delivery options to our telecommand we want to send.
```
[the remark this answers is not in the document]
Comment (fussel178) [reply docs/client/tutorials/301-writing-a-stateful-dialog.mdx:64]: Not here :package:

[the remark this answers is not in the document]
Comment (fussel178) [reply docs/client/tutorials/301-writing-a-stateful-dialog.mdx:64]: Done in [`a4df0f4` (#160)](https://github.com/wuespace/telestion-docs/pull/160/commits/a4df0f4657de47e4ebf85dbf3d3f655867cf0731)

File: docs/client/tutorials/300-writing-a-static-dialog.mdx [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+	);
+}
+```
+
+The `showDialog` function needs an unique id across the Web Client and a configuration object.
+
+The object is the main part of the dialog.
+It defines the content of the different parts in the dialog.
+For example, the `title` property defines the content in the upper left corner
+or the `content` property the center part of the dialog.
```
[the remark this answers is not in the document]
Comment (fussel178) [reply docs/client/tutorials/300-writing-a-static-dialog.mdx:316]: Done in [`a4df0f4` (#160)](https://github.com/wuespace/telestion-docs/pull/160/commits/a4df0f4657de47e4ebf85dbf3d3f655867cf0731)

File: docs/client/tutorials/300-writing-a-static-dialog.mdx [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+			<Flex direction="column" gapsize="size-100">
+				<TextField label="Telecommand" width="100%" />
+				<ActionButton>Send</ActionButton>
+			</Flex>
+		</View>
+	);
+}
+```
+
+If you start the Web Client and add the widget to your dashboard you should see something like:
```
[the remark this answers is not in the document]
Comment (fussel178) [reply docs/client/tutorials/300-writing-a-static-dialog.mdx]: Done in [`a4df0f4` (#160)](https://github.com/wuespace/telestion-docs/pull/160/commits/a4df0f4657de47e4ebf85dbf3d3f655867cf0731)

File: docs/client/tutorials/300-writing-a-static-dialog.mdx [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+  keeping the "step by step" flow going)
+-->
+
+<!-- Relevant imports: -->
+
+import { Reference, Image } from '/components';
+
+<!-- Short description of what the article covers: -->
+
+In this tutorial, we write a static dialog for a widget in the web client.
```
[the remark this answers is not in the document]
Comment (fussel178) [reply docs/client/tutorials/300-writing-a-static-dialog.mdx:52]: Done in [`a4df0f4` (#160)](https://github.com/wuespace/telestion-docs/pull/160/commits/a4df0f4657de47e4ebf85dbf3d3f655867cf0731)

**Review by pklaschka — APPROVED** [review]
LGTM!!

[cool](https://graphite-user-uploaded-assets.s3.amazonaws.com/default-memes/cool.gif)

**Review by pklaschka — APPROVED** [review]

**Pull request merged**